### ISSUE TITLE
Consider OSF or internet archive for storage or archive of ontologies

### ISSUE THREAD

**Issue by cmungall** [OBOFoundry/OBOFoundry.github.io#494]
* many groups use github for managing release files. Works well with zenodo, which is backed by CERN
 * some files are too big for github
 * not everyone takes advantage of this. Some groups do not preserve older versions which is not good. E.g. CHEBI

It would be good to archive all versioned PURLs, and also offer a solution for people to release their files that doesn't require large derived owl files checked into github

OSF is one possibility: https://osf.io/
Guaranteed for 50 years. And we are very much in line with their mission.

There is a nice python API and CLI developed by Titus Brown's lab: http://osfclient.readthedocs.io/

Another option is https://git-annex.branchable.com/tips/Internet_Archive_via_S3/, @jhpoelen is a big fan of this approach

**Comment by jhpoelen** [comment]
What I like about https://git-annex.branchable.com is that you can manage multiple copies across all sorts of different storage - thumbdrives, internet archive, s3. I like the idea of not putting all my eggs into one basket and being able to (automatically) migrate data from one place to another.

**Comment by cmungall** [comment]
@jhpoelen - thanks for the tip - do you have any pointers to docs where you've used this?

**Comment by cmungall** [comment]
I've played a little with both. One requirement from an OBO pov is that we don't want to have to map folders full of produces to SHA-type URLs on a per-file basis.

This seems to be a current limitation of OSF. If I make a project that gets assigned an id `xyz12`, and then I add a file imports/foo_imports.owl then that file will get some id like `qwrt5`. Ideally we'd have a URL like `xyz12/imports/foo_imports.owl`, otherwise it gets complicated with PURL configuration. I've written to ask them if this is something they support.

I have played a bit with git annex but still learning, it seems as if each file gets assigned a SHA type URL.

**Comment by jhpoelen** [comment]
As far as I understand, the hashes used to store stuff on the blob systems and git is used for names. So, you can have a folder structure with symlinks in git and use git-annex to materialize (or inflate) symlinks locally from whatever source it is available.

**Comment by jhpoelen** [comment]
The example at https://github.com/globalbioticinteractions/archive is a bit confusing, because the names of the symlinks happen to be sha 256 hashes also. E.g., https://github.com/globalbioticinteractions/archive/tree/master/datasets/globalbioticinteractions/natural-history-museum-london-interactions-bank .

**Comment by alanruttenberg** [comment]
@cmungall re: xyz12/imports/foo_imports.owl

I don't see that we need purls like that. All we need is that during release all imports get dated purls. The source files are uploaded and then all the purls are redirected to their sha-based URL. We don't need to rewrite the purls that are to be imported after build. All we need is a mapping from the files uploaded to the OSF urls.  Or am I misunderstanding you? 

The code I have for IAO release does this, including making a local dated copy of any undated imports.

**Comment by cmungall** [comment]
the purls would be of the form `$OBO/{ontid}/imports/foo_imports.owl`. Each individual file is given a new SHA-type URL. The desire is that their native URLs would be of the form `osf.io/{projectId}/imports/foo_imports.owl` to make mapping more straightforward. But this less important when I complete this PR: https://github.com/dib-lab/osf-cli/pull/119

**Comment by david4096** [comment]
Please check out https://github.com/ipfs/ipfs .

You can use their naming system to maintain a single hash that will keep the latest version of your ontology.

**Comment by nlharris** [comment]
What is the status of this?

**Comment by nlharris** [comment]
See also #753

**Comment by cmungall** [comment]
we need to update this ticket with recent things we have learned

some ontologies are now using gh releases for larger files. This is easy to do in python

@cthoyt has a handy python script for using zenodo
https://github.com/pyobo/pyobo/blob/master/src/pyobo/zenodo_client.py

but zenodo may be best for archive rather than live serving up latest ontology

**Comment by cthoyt** [comment]
> we need to update this ticket with recent things we have learned
> 
> some ontologies are now using gh releases for larger files. This is easy to do in python
> 
> @cthoyt has a handy python script for using zenodo
> https://github.com/pyobo/pyobo/blob/master/src/pyobo/zenodo_client.py
> 
> but zenodo may be best for archive rather than live serving up latest ontology

In fact I just made it into its own package today but unfortunately PyPI's backend has minor outages and I haven't been able to upload it yet. See: https://github.com/cthoyt/zenodo-client/

I will have a few examples of it being used in the wild soon. It also needs a feature that lets you deal with the situation where a deposition ID is not yet available, since it's a bit clunky there

**Comment by matentzn** [comment]
We used to have this issue with Zenodo that there is no way to refer to a specific file in the _latest_ release. Is that still the case?

**Comment by matentzn** [comment]
Yeah, and @kltm kept reminding us that an archive is an archive and _not_ a fileserver :P So using Zenodo as a fileserver is sort of abuse anyways..

**Comment by cmungall** [comment]
relevant https://dvc.org/doc/start/data-and-model-versioning

h/t @realmarcin

**Comment by kltm** [comment]
Self-tagging @kltm

**Comment by jhpoelen** [comment]
Ever considering using a more flexible content-based addressing approach like https://github.com/bio-guoda/preston ? Name/location-based single parent structures to organize content seems at odds when dealing with rich semantics supported through OBO Foundry's member ontologies.
